src/main/java/subway/line/LineService.java

```diff
@@ -0,0 +1,59 @@
+package subway.line;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import subway.station.Station;
+import subway.station.StationRepository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+    private LineRepository lineRepository;
+    private StationRepository stationRepository;
+
+    public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Station up = createStationById(lineRequest.getUpStationId());
+        Station down = createStationById(lineRequest.getDownStationId());
+        Line line = lineRepository.save(new Line(lineRequest.getName(), lineRequest.getColor(), up.getId(), down.getId(), lineRequest.getDistance()));
+        return createLineResponse(line);
+    }
+
+    public List<LineResponse> findAllLines() {
+        return lineRepository.findAll().stream()
+                .map(this::createLineResponse)
+                .collect(Collectors.toList());
+    }
+
+    public LineResponse findLineById(Long id) {
+        Line line = lineRepository.findById(id).orElseThrow(() -> new IllegalArgumentException("존재하는 노선이 없음 : " + id));
+        return createLineResponse(line);
+    }
+
+    @Transactional
+    public void deleteLineById(Long id) {
+        lineRepository.deleteById(id);
+    }
+
+    private LineResponse createLineResponse(Line line) {
+        return new LineResponse(
+                line.getId(),
+                line.getName(),
+                line.getColor(),
+                List.of(createStationById(line.getUpStationId()), createStationById(line.getDownStationId()))
+        );
+    }
+
+    private Station createStationById(Long id) {
+        Station station = stationRepository.findById(id).orElseThrow(() -> new IllegalArgumentException("역이 존재하지 않음 : " + id));
```

Comment:
[객체지향생활체조](https://edu.nextstep.camp/s/ZlcCZQ2Y/ls/USFEu0qR)

> 규칙 4: 한 줄에 점을 하나만 찍는다.

```suggestion
        Station station = stationRepository.findById(id)
            .orElseThrow(() -> new IllegalArgumentException("역이 존재하지 않음 : " + id));
```
[Google Style Guide](https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break) 참고해보시면 좋을 것 같아요 😄